.github/workflows/main.yml

```diff
@@ -67,14 +70,36 @@ jobs:
       - name: Test
         run: yarn test
 
+      - name: Coverage
+        run: yarn coverage
+
+      - name: Check test coverage
+        id: test-coverage
+        uses: johanvanhelden/gha-clover-test-coverage-check@v1
+        with:
+          percentage: '65'
+          filename: './coverage/clover.xml'
+
+      - name: 'Export coverage results'
+        uses: 'codecov/codecov-action@v1'
+        with:
+          file: './coverage/clover.xml'
+          env_vars: NODE_VERSION
+
+      - name: 'Archive code coverage results'
+        uses: 'actions/upload-artifact@v2'
+        with:
+          name: 'NODE ${{ matrix.node-version }}'
+          path: './coverage/clover.xml'
```

Comment:
I think that the coverage job should be done only one time (node 16) not in matrix.